ecl/hthor/hthor.cpp

```diff
@@ -4929,7 +4953,7 @@ bool CHThorSelfJoinActivity::fillGroup()
     failingOuterAtmost = false;
     OwnedConstRoxieRow next;
     unsigned groupCount = 0;
-    next.setown(input->nextInGroup());
+    next.setown(groupedInput->nextInGroup());
```

Comment:
It seems to be the same in Roxie - fillGroup is not called in the limited case I think.